synthesizer/src/vm/verify.rs

```diff
@@ -134,6 +134,37 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
 
         lap!(timer, "Check for duplicate elements");
 
+        // Get the consensus version.
+        let consensus_version = N::CONSENSUS_VERSION(self.block_store().current_block_height())?;
+
+        // Acquire a read lock on the process.
+        let process = self.process.read();
```

Comment:
We can consider minimizing this lock scope.
This would be safe since editions are always incremented and a write lock is held when the process stacks are updated during finalization.

Reply:
I think we can leave it. Or if you insist, move it just inside of `if let Transaction::Execute(id, _, execution, _) = transaction {`. There's a minor efficiency gain from taking it just once for all transitions.